### DISCUSSION TITLE
Include pure rust scripts in release

### DISCUSSION THREAD

**Discussion by mlucool** [PyO3/maturin#1006] [Q&A] [answered]
Hi,

I have a pure rust script that I'd like to include in my release. What is the right way to do this? I see https://maturin.rs/metadata.html#add-console-scripts, but that created a console script (i.e. with python binding). Per python-packaging.com:
> This has advantage of being generalizeable to non-python scripts, as well: funniest-joke could be a shell script, or something completely different.

Thanks!

**Comment by messense** [comment, marked as answer]
What's your current setup? 

If you just want to distribute a Rust binary, you can use [`bin` bindings](https://maturin.rs/bindings.html#bin). But if you want to distribute a Python extension and a Rust binary that's not implemented yet, see #368.

**Reply by mlucool** [reply]
Thanks! I was hoping to do both but I can work around this.

**Comment by anuradhawick** [comment]
@mlucool  would you mind sharing your workaround?

I have a workspace few crates one of which is a binary and one is the python wrapper. Not sure how to include the bin file from the binary crate.

Thanks

**Reply by mlucool** [reply]
For this project I only needed the python bindings for testing testing my rust code. See: https://github.com/deshaw/nbstripout-fast

**Reply by anuradhawick** [reply]
Thanks for sharing that, I ended up wrapping the main as a py function and added a script to call it; basically a CLI app that way.

**Reply by anuradhawick** [reply]
In case you were interested;

Proj: https://github.com/anuradhawick/kmertools
Cli: https://github.com/anuradhawick/kmertools/blob/main/pip/src/lib.rs
Pyproject toml: https://github.com/anuradhawick/kmertools/blob/main/pip/pyproject.toml
